src/Metal/MTLDevice.cs

```diff
@@ -42,6 +42,28 @@ public static partial class MTLDevice {
 				return system_default;
 			}
 		}
+
+#if MONOMAC
+		//[Mac (10,13, onlyOn64: true), NoiOS, NoWatch, NoTV]
+		//[DllImport (Constants.MetalLibrary)]
+		//static extern IMTLDevice[] MTLCopyAllDevicesWithObserver ([NullAllowed] out NSObject observer, MTLDeviceNotificationHandler handler);
```

Comment:
I see you're defining the delegate in metal.cs, but there's no API in metal.cs that uses the delegate. I wonder if the generator is being smart and not copying the delegate to the generated code because it's not being used. Maybe try moving the delegate to Metal/MTLDevice.cs?